src/rttr/detail/misc/standard_types.cpp

```diff
@@ -90,7 +90,8 @@ RTTR_REGISTRATION
                 .method("operator[]",   rttr::select_overload<char&(size_t)>(&std::string::operator[]))
                 .method("operator[]",   rttr::select_non_const(&std::string::operator[]))
 #endif
-#if __cplusplus >= 201703L || (RTTR_COMPILER == RTTR_COMPILER_MSVC && !defined(RTTR_NO_CXX17_NOEXCEPT_FUNC_TYPE))
+#if __cplusplus >= 201703L || (RTTR_COMPILER == RTTR_COMPILER_MSVC && !defined(RTTR_NO_CXX17_NOEXCEPT_FUNC_TYPE)) || \
+     (RTTR_COMP_VER >= 900 && RTTR_PLATFORM == RTTR_PLATFORM_APPLE)
```

Comment:
Just out of curiosity: Why does that fix on macOs? is there no string::data method? Strange =)